### DISCUSSION TITLE
How to run Conductor Server 3.0.0+ programmatically? (Migration)

### DISCUSSION THREAD

**Discussion by DPain** [Netflix/conductor#2417] [Q&A]
So prior to Conductor Server 3.0, I've been running Conductor Server with a Spring Boot wrapper project that would configure various settings prior and then fire up a thread that would basically call Conductor Server's Main method located in `com.netflix.conductor.bootstrap.Main` to start up the Conductor Server with necessary settings already configured.

Since Conductor Server changed to Spring Boot, I can no longer access Conductor Server's equivalent main method for v3.0.0+ which is in `com.netflix.conductor.Conductor`. I don't have much knowledge about Spring Boot, but I assume its due to Conductor Server using Spring Boot.

How can I access the main method of Conductor Server 3.0.0+? After further research I found that the approach I took before can no longer be done as per: `https://stackoverflow.com/questions/40089443/how-to-add-a-dependency-to-a-spring-boot-jar-in-another-project` If so, which approach would you recommend me to do?

Thanks a bunch in advance :)

**Comment by kishorebanala** [comment]
@DPain Can you share an example of how you were configuring this earlier? Current Spring boot implementation doesn't do much, and you should be able to configure Spring boot the way you want. Having a concrete example of what you are trying to do helps.

**Reply by DPain** [reply]
Sorry for the late response!

A quick example would be something like this.
```
import com.netflix.conductor.bootstrap.Main;

@SpringBootApplication
@Controller
@EnableAutoConfiguration(exclude={Some.class})
public class ConductorBoot {
	
	public static void main(String[] args) throws InterruptedException {
        SpringApplication app = new SpringApplication(ConductorBoot.class);
        app.run(args);
	}
	
	@EventListener(ApplicationReadyEvent.class)
	public void startConductorServer() throws Exception {
		// Spring Boot app fetches necessary values that need to be fed into Conductor Server.
		// Values are fetched during runtime.
		setEnvVarDynamically()

		Main.main();
	}

}
```

Since Conductor server 3.0.0+ is now a Spring Boot application, I am no longer able to fetch the main method to run it inside the application. I was wondering if there was an alternative.

**Reply by aravindanr** [reply]
@DPain The SpringBoot wrapper is not necessary anymore as Conductor itself uses SB directly. But if you need to, this might work,

```java
public class ConductorBoot {	
	public static void main(String[] args) throws InterruptedException {
        SpringApplication app = new SpringApplication(Conductor.class);
        app.run(args);
	}
}
```

**Reply by DPain** [reply]
Thanks, I'll try that!

**Reply by DPain** [reply]
To import the Conductor class that's in conductor-server, do I need to add anything else in my pom.xml file?
I literally added
```
<dependency>
	<groupId>com.netflix.conductor</groupId>
	<artifactId>conductor-server</artifactId>
	<version>3.3.4</version>
	<classifier>boot</classifier>
</dependency>
```

But I can't seem to access `com.netflix.conductor.Conductor` in my codebase, but I can access everything else like `com.netflix.client.*` and more. I do see `com.netflix.conductor:conductor-server:jar:boot:3.3.4:compile` and all the other conductor libraries underneath conductor server when I run `mvn dependency:tree`.

**Reply by DPain** [reply]
Ended just building the Conductor server myself and adding the necessary features myself. Seemed like that'd be less the hassle.
